cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/ApplicationVerificationStage.java

```diff
@@ -109,7 +109,12 @@ protected void verifyData(Id.Application appId,
                             ApplicationSpecification specification) throws DatasetManagementException {
     // NOTE: no special restrictions on dataset module names, etc
     VerifyResult result;
+    MacroChecker macroChecker = new MacroChecker();
     for (DatasetCreationSpec dataSetCreateSpec : specification.getDatasets().values()) {
+      if (macroChecker.isMacro(dataSetCreateSpec.getInstanceName())) {
```

Comment:
this logic should not be in the cdap platform. Macros are a concept limited to a PluginConfig. Making sure you don't create a dataset with a macro in it is the responsibility of the plugin.